doc/compressed_transactions.md

```diff
@@ -0,0 +1,153 @@
+the Txid if the transaction input is at least one hundred blocks old.
+
+## 3 Schema
+
+### 3.1 Primitives
+
+| Name             | Width     | Description |
+|------------------|-----------|-------------|
+| VarInt           | 1-5 Bytes | For 0-253, encode the value directly in one byte. For 254-65535, encode 254 followed by 2 little-endian bytes. For 65536-(2^32-1), encode 255 followed by 4 little-endian bytes. This is the same varint used to length-prefix strings in the ordinary transaction encoding. |
+| VarInt Flag      | 2 Bits    | 1, 2 or 3 indicate literal values. 0 indicates that the value will be encoded in a later VarInt. |
+| Bitstream VarInt | Variable  | 7-bit little-endian encoding, with each 7-bit word encoded in a byte. The highest bit of each byte is 1 if more bytes follow, and 0 for the last byte. |
+| VLP-Bytestream   | Variable  | A VarInt Length Prefixed Bytestream, Has a VarInt prefixed to determine the length. |
+
+### 3.2 General Schema
+
+| Name                  | Width           | Description |
+| ----------------------|----------------|-------------|
+| Transaction Metadata  | 1-2 Bytes      | Information on the structure of the transaction. See section 3.2. |
+| Minimum Blockheight | Bitstream VarInt | Transaction locktime and input blockheights are given as offsets from this value. |
+| Version               | Variable       | If the Version was greater then 3 then this is an optional VarInt containing the Version. |
+| Input Count VarInt    | Variable       | If the Input Count was greater then 3 then this is an optional VarInt containing the Input Count. |
+| Output Count VarInt   | Variable       | If the Output Count was greater then 3 then this is an optional VarInt containing the Output Count. |
+| LockTime VarInt       | Variable       | An optional VarInt containing the LockTime if its non zero. |
+| Input/Output Metadata | Variable       | A variable-length bitstream padded to a multiple of 8 bits. Contains metadata on the Inputs and Outputs of the Transaction. See section 3.3 and 3.4. |
+| Input Data            | Variable       | For each Input. See Section 3.5. |
+| Output Data           | Variable       | For each Output. See Section 3.6. |
+
+For the four VarInts listed above we could use a more compact bit encoding for these but they are already a fall back for the bit encoding of the Transaction Metadata.
+
+### 3.3 Transaction Metadata
+
+| Name         | Width       | Description |
+| -------------|-------------|-------------|
+| Version      | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction version. |
+| Input Count  | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction input count. |
+| Output Count | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction output count. |
+| LockTime     | 1 bit       | Wether the LockTime has a value. |
```

Comment:
Also, if the locktime *doesn't* have a value we should specify what its value is. e.g. `0 if the locktime is zero; otherwise indicates a future VarInt`

Reply:
Fixed, Thanks!